pkg/nats/stream/producer_client.go

```diff
@@ -51,33 +69,34 @@ func (pc *ProducerClient) AddStream(
 	pc.mu.Lock()
 	defer pc.mu.Unlock()
 
+	if _, ok := pc.activeConsumers[consumerID]; !ok {
+		pc.activeConsumers[consumerID] = consumerInfo{
+			HeartbeatRequestSub: heartBeatRequestSub,
+			ActiveStreamInfo:    make(map[string]StreamInfo),
+		}
+	}
+
+	if _, ok := pc.activeConsumers[consumerID].ActiveStreamInfo[streamID]; ok {
+		return fmt.Errorf("cannot create request with same streamId %s again", streamID)
+	}
+
 	streamInfo := StreamInfo{
 		ID:         streamID,
 		RequestSub: requestSub,
 		CreatedAt:  time.Now(),
 		Cancel:     cancelFunc,
 	}
 
-	if pc.activeStreamInfo[consumerID] == nil {
-		pc.activeStreamInfo[consumerID] = make(map[string]StreamInfo)
-	}
-
-	if _, ok := pc.activeStreamInfo[consumerID][streamID]; ok {
-		return fmt.Errorf("cannot create request with same streamId %s again", streamID)
-	}
-
-	pc.activeStreamInfo[consumerID][streamID] = streamInfo
-	pc.activeConnHeartBeatRequestSubjects[consumerID] = heartBeatRequestSub
-
+	pc.activeConsumers[consumerID].ActiveStreamInfo[streamID] = streamInfo
 	return nil
 }
 
 func (pc *ProducerClient) RemoveStream(consumerID string, streamID string) {
 	pc.mu.Lock()
 	defer pc.mu.Unlock()
 
-	activeStreamIdsForConn, ok := pc.activeStreamInfo[consumerID]
-	if !ok {
+	activeStreamIdsForConn := pc.activeConsumers[consumerID].ActiveStreamInfo
+	if activeStreamIdsForConn == nil {
 		return
 	}
```

Comment:
can this return nil pointer exception if no entry for for `consumerID` is found?